design/decouple-ceph-version.md

```diff
@@ -0,0 +1,158 @@
+
+## Coupled (Legacy) Design
+
+The approach of embedding Ceph into the Rook image had several advantages that contributed to the design.
+- Simpler development and test matrix. A consistent version of Ceph is managed and there are no unstable Ceph bits running in the cluster.
+- Simpler upgrade path. There is only one version to worry about upgrading.
+
+The project is growing out of these requirements and we need to support some added complexity in order to get the benefits of the decoupled versions.
+
+## New Design
+
+There are two versions that will be specified independently in the cluster: the Rook version and the Ceph version.
+
+### Rook Version
+
+The Rook version is defined by the operator's container `image` tag. All Rook containers launched by the operator will also launch the same version of the Rook image.
+The full image name is an important part of the version. This allows the container to be loaded from a private repo if desired.
+
+In this example, the Rook version is `rook/ceph:v0.8.1`.
+
+```yaml
+apiVersion: apps/v1beta1
+kind: Deployment
+metadata:
+  name: rook-ceph-operator
+spec:
+  template:
+    spec:
+      containers:
+      - name: rook-ceph-operator
+        image: rook/ceph:v0.8.1
+```
+
+### Ceph Version
+
+The Ceph version is defined as the property `cephImage` in the Cluster CRD. All Ceph daemon containers launched by the Rook operator will use this image, including the mon, mgr,
+osd, rgw, and mds pods. The significance of this approach is that the Rook binary is not included in the daemon containers. All initialization performed by Rook to generate the Ceph config and prepare the daemons must be completed in an [init container](https://github.com/rook/rook/issues/2003). Once the Rook init containers complete their execution, the daemon container will run the Ceph image. The daemon container will no longer have Rook running.
+
+In the following Cluster CRD example, the Ceph version is Mimic 13.2.1.
+
+```yaml
+apiVersion: ceph.rook.io/v1beta1
+kind: Cluster
+metadata:
+  name: rook-ceph
+  namespace: rook-ceph
+spec:
+  cephImage: ceph/ceph:v13.2.1
+```
+
+### Operator Requirements
+
+The operator needs to run the Ceph client tools to manage the cluster. For example, the `ceph` tool is needed for general Ceph configuration and status, while `radosgw-admin` is required for managing an object store. Therefore, all the necessary client tools will still be included in the Rook image.
+
+The client tools are tested by the Ceph team to be backward and forward compatible by two versions. Therefore, the operator can support a version of Ceph up to two versions older than the client tools it contains.
```

Comment:
Note: In general as we currently mostly rely on parsing command outputs we should improve this to use a (stable) API instead of commands directly, see #2088.